lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -669,7 +669,7 @@ func sufficientHints(nHintsLeft int, currentAmount,
 		return true
 	}
 
-	if currentAmount >= targetAmount {
+	if targetAmount != 0 && currentAmount >= targetAmount {
```

Comment:
Since this has privacy implications (zero amount potentially leaking more unannounced channels than setting a specific amount), should we add a comment to both the CLI `--amt` flag and the `value`/`value_msat` fields on the RPC?

Reply:
Yep, that wouldn't hurt. I've just pushed an extra commit adding those comments into the `private` field instead, to avoid the repetition in the `value/value_msat` ones. Let me know if that works.